clients/object_store_client.go

```diff
@@ -29,10 +39,27 @@ func UploadToOSURL(osURL, filename string, data io.Reader) error {
 		return fmt.Errorf("failed to parse OS URL %q: %s", osURL, err)
 	}
 
-	_, err = storageDriver.NewSession("").SaveData(context.Background(), filename, data, nil, 30*time.Second)
+	writeOperation := makeOperation(func() error {
+		_, err := storageDriver.NewSession("").SaveData(context.Background(), filename, data, nil, 30*time.Second)
+		return err
+	})
+
+	err = backoff.Retry(writeOperation, backoff.WithMaxRetries(backOff, 2))
 	if err != nil {
 		return fmt.Errorf("failed to write file %q to OS URL %q: %s", filename, osURL, err)
 	}
 
 	return nil
 }
+
+func newBackOffExecutor() *backoff.ExponentialBackOff {
+	backOff := backoff.NewExponentialBackOff()
+	backOff.InitialInterval = 200 * time.Millisecond
+	backOff.MaxInterval = 1 * time.Second
+
+	return backOff
+}
+
+var makeOperation = func(fn func() error) func() error {
+	return fn
+}
```

Comment:
This is mainly to make the retryable callback mockable. Otherwise i would need to mock the object store session to return errors for tests. This seemed the fastest solution to me